metaseq/trainer.py

```diff
@@ -1176,6 +1189,38 @@ def is_consistent(tensor):
     def _reduce_and_log_stats(
         self, logging_outputs, sample_size, grad_norm=None, ewm_loss_ratio=0
     ):
+        # perform a bunch of arch-specific gradient metrics
+        for name, param in self.model.named_parameters():
+            if (not self.is_fsdp) or self.dont_log_param_and_grad_norm:
+                break
+            if param.grad is None:
+                continue
+            nice_name = name.replace("module._fsdp_wrapped_module._fpw_module.", "")
+            nice_name = nice_name.replace("_fsdp_wrapped_module._fpw_module.", "")
+            nice_name = nice_name.replace("._fsdp_wrapped_module.flat_param_0", "")
+            nice_name = nice_name.replace("decoder.layers.", "layer")
+            # threshold for near zeros
+            threshold = torch.finfo(param.grad.dtype).tiny * 2
```

Comment:
This threshold seems really low? Do we not want to check for things close to zero as well?

Reply:
ah this was the threshold during FP16 OPT days... keeping it around for now but sufficiently close to 0 here would be good either way